src/libraries/System.Private.CoreLib/src/System/ArgumentNullException.cs

```diff
@@ -50,5 +52,16 @@ public ArgumentNullException(string? paramName, string? message)
         protected ArgumentNullException(SerializationInfo info, StreamingContext context) : base(info, context)
         {
         }
+
+        /// <summary>Throws an <see cref="ArgumentNullException"/> if <paramref name="argument"/> is null.</summary>
+        /// <param name="argument">The reference type argument to check for nullability.</param>
+        /// <param name="paramName">The name of the parameter with which <paramref name="argument"/> corresponds.</param>
+        public static void ThrowIfNull([NotNull] object? argument, [CallerArgumentExpression("argument")] string? paramName = null)
+        {
+            if (argument is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(paramName);
```

Comment:
For my education, why go through ThrowHelper here? Why not just
```suggestion
                throw new ArgumentNullException(paramName);
```

Reply:
Because then either a) the IL size increases enough that the JIT decides not to inline the check,  or much more likely b) it does inline it and now every call site is paying for the asm for the `throw new ANE(name)` that also gets inlined.

Reply:
Ah, got it. We want the check to be inlined, but not the throw. Thanks for the explanation.

Reply:
> We want the check to be inlined, but not the throw

Exactly